aspnet/publishing/iis.rst

```diff
@@ -1,116 +1,178 @@
+Set the **.NET CLR version** to **No Managed Code**.
 
-Supported operating systems
------------------------------------
+	.. image:: pubIIS/_static/editapppool.png
 
-The following operations systems are supported:
+Create a Data Protection Registry Hive
+--------------------------------------
 
-- Windows 7 and newer
-- Windows 2008 R2 and newer
+Data Protection keys used by ASP.NET applications are stored in registry hives external to the applications. To persist the keys for a given application, you must create a registry hive for the application's application pool.
+
+For standalone IIS installations, you may use the `Data Protection Provision-AutoGenKeys.ps1 PowerShell script <https://github.com/aspnet/DataProtection/blob/dev/Provision-AutoGenKeys.ps1>`__ for each application pool used with an ASP.NET Core application. The keys will be persisted in the registry.
+
+In web farm scenarios, an application can be configured to use a UNC path to store its data protection key ring. By default, the data protection keys are not encrypted. You can deploy an x509 certificate to each machine to encrypt the key ring. See :ref:`Configuring Data Protection <data-protection-configuring>` for details.
+
+.. include:: ./dataProtectionWarning.txt
+
+Common Errors
+-------------
+
+The following is not a complete list of errors. Should you encounter an error not listed here, please leave a detailed error message in the DISQUS section below (click `Show comments` to open the DISQUS panel).
+
+To diagnose problems with IIS deployments, study browser output, examine the server's **Application** log through **Event Viewer**, and `enable logging <https://docs.asp.net/en/latest/fundamentals/logging.html>`__ in the application. The **ASP.NET Core Module** log will be found on the path provided in the `stdoutLogFile` attribute of the `\<aspNetCore\>` element in `web.config`. Any folders on the path provided in the attribute value must exist in the deployment. You must also set `stdoutLogEnabled="true"` to enable application logging. Applications that use the `publish-iis` tooling to create the `web.config` file will default the `stdoutLogEnabled` to `false`, so you must manually provide the file or modify the file in order to enable application logging.
+
+A quick way to determine if the IIS reverse proxy to the Kestrel server is working properly is to perform a simple static file request (e.g., a stylesheet, script, or image) from the application's static assets (i.e., those in `wwwroot`) using `Static File Middleware <https://docs.asp.net/en/latest/fundamentals/static-files.html>`__. If the application can serve static files but MVC Views and other endpoints are failing, the problem is less likely related to the IIS-ASP.NET Core Module-Kestrel configuration and more likely within the application itself (e.g., MVC routing, 500 Internal Server Error, etc.). In most cases, enabling application logging will assist in troubleshooting problems within the application.
+
```

Comment:
The reason I included this is that it quickly assesses the status of the IIS-module leg of the config. I use this test quite a lot when I'm helping devs in a chat situation (e.g., JabbR), and it quickly puts us in the right direction to go with troubleshooting.

> tell them to deploy something very simple

This is also very good as a general suggestion, but I'm not sure how to work it in as general advice into a "publishing to IIS" doc. I have told devs who show up with very broken projects to take a step back with a cli-sample or IISIntegration sample and get that running, then go forward with their mega-monster app.